packages/ui/src/App.svelte

```diff
@@ -38,17 +39,43 @@
       dispatch('tab-change', tab);
     };
 
+    export let initialOpts: InitialOptions = {};
+    let initialValuesSet = false;
+
     let allOpts: { [k in Kind]?: Required<KindedOptions[k]> } = {};
     let errors: { [k in Kind]?: OptionsErrorMessages } = {};
 
-    let contract: Contract = new ContractBuilder('MyToken');
+    let contract: Contract = new ContractBuilder(initialOpts.name ?? 'MyToken');
 
     $: functionCall && applyFunctionCall()
 
     $: opts = allOpts[tab];
 
     $: {
       if (opts) {
+        if (!initialValuesSet) {
+          switch (opts.kind) {
+            case 'ERC20':
+              opts.name = initialOpts.name ?? opts.name;
+              opts.symbol = initialOpts.symbol ?? opts.symbol;
+              opts.premint = initialOpts.premint ?? opts.premint;
+              break;
+            case 'ERC721':
+              opts.name = initialOpts.name ?? opts.name;
+              opts.symbol = initialOpts.symbol ?? opts.symbol;
+              break;
+            case 'ERC1155':
+              opts.name = initialOpts.name ?? opts.name;
+              break;
+            case 'Governor':
+              opts.name = initialOpts.name ?? opts.name;
+              break;
+            case 'Custom':
+              opts.name = initialOpts.name ?? opts.name;
+              break;
+          }
+          initialValuesSet = true;
+        }
```

Comment:
Maybe cleaner for now? Feel free to refactor or ignore, I just noticed name is set many times.

```suggestion
        if (!initialValuesSet) {
          opts.name = initialOpts.name ?? opts.name;
          switch (opts.kind) {
            case 'ERC20':
              opts.premint = initialOpts.premint ?? opts.premint;
            case 'ERC721':
              opts.symbol = initialOpts.symbol ?? opts.symbol;
              break;
            case 'ERC1155':
            case 'Governor':
            case 'Custom':
          }
          initialValuesSet = true;
        }
```